### PR TITLE
Distinguish SASS and CSS

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request makes SASS a language of its own, distinct from CSS.
This change was requested several times in #2933, #3084, #2585 and #2650.

### CODE REVIEW

**Comment by arfon** [comment]
...so if we do this then I think we need to consider making the same change for `HAML`, `Handlebars` etc. and many of the other languages that are currently grouped into some higher-level category.

My rationale for *not* doing this is that SASS is almost always used to generate CSS (please correct me if I'm wrong here) and so it makes sense (to me at least!) to have this listed under CSS for the general repository stats.

I would also note that this change doesn't affect search results/[GitHub explore](https://github.com/trending) (trending repos by language). Languages such as SASS are already discoverable there.

Basically we can consider doing this if we really want to but it should be part of a much larger rethink of what Linguist languages groups are.

**Comment by controversial** [comment]
@arfon You're right that `sass` is almost always used to generate CSS. However, it has its own syntax which is quite different from that of CSS, unlike the other language in the sass family, scss. Isn't your argument also true of many compiled languages? CoffeeScript and JavaScript are separate in Github's eyes. SASS is to CSS as CoffeeScript is to JavaScript. It's about the same level of separation.

**Comment by arfon** [comment]
> Isn't your argument also true of many compiled languages? CoffeeScript and JavaScript are separate in Github's eyes. SASS is to CSS as CoffeeScript is to JavaScript. It's about the same level of separation.

It is, and I want to be clear that I'm not completely against this idea, I just think we need to make sure we've thought this through before we start removing `group:` files from `languages.yml`.

@controversial I'd be curious to hear what problem you're trying to solve with this change? Is this a UI concern of the language bar on the repo or something else?

**Comment by controversial** [comment]
@arfon That's basically it, I'd like my repo to show that it's styled with SASS instead of CSS. It's not a big deal, but I think it'd be better. ¯\_(ツ)_/¯

**Comment by Alhadis** [comment]
If CoffeeScript is recognised as its own language, then Sass should too. Otherwise, we may as well be categorising the former as "JavaScript" as well.

**Comment by controversial** [comment]
Please?

**Comment by arfon** [comment]
> @arfon That's basically it, I'd like my repo to show in the top bar that it's styled with SASS instead of CSS. It's not a big deal, but I think it'd be better. ¯_(ツ)_/¯

Sorry for the delayed reply here. I'm not comfortable merging this until we've taken a good look at _all_ grouped languages. It's possible there is UI fix on GitHub that would address this more satisfactorily.

**Comment by controversial** [comment]
Ok. Makes sense, I guess.

**Comment by nazar-pc** [comment]
Basically, the same story as #2978, where Jade/Pug is definitely not HTML

**Comment by controversial** [comment]
Any update? I'm once again back here after discovering that Slim and Jade are classified as HTML, not as separate languages. Seriously think this is a necessary change.

@arfon you say there might be a UI fix, but that sounds like something that's far, far less likely to happen, especially not in the short term. For now, do you still have a problem with separating out similar languages? This issue has been brought up many many times about different languages, and I haven't yet seen a very compelling argument for keeping them together beyond "there might be a UI fix."

You say
> until we've taken a good look at *all* grouped languages

what does this entail? I'd really like to see some change soon.

**Comment by EmmaRamirez** [comment]
I feel like this PR should really be considered. SASS is quite a different extension of CSS.

It's also not particularly consistent--why is Smarty, for example, considered separate from PHP?

**Comment by ruchern** [comment]
Is this going to happen afterall?

**Comment by yegortimoshenko** [comment]
I'm not sure why there's grouping at all in the user interface, GitHub could just show each language as its own. Some grouping can be done if several files in group take very small percentage of repo's source code (as done with "Other"), but it'd be neat to distinct various preprocessors from the language they compile to.

**Comment by plibither8** [comment]
@arfon @Alhadis is it going to happen? It's been more that two years since the PR was opened

**Comment by pchaigno** [comment]
Arfon is not working on this project anymore, but as he said:

> I'm not comfortable merging this until we've taken a good look at _all_ grouped languages

I agree with him on this. But it takes a lot of time to do this kind of work.

**Comment by plibither8** [comment]
This might sound rather curt but I feel that two years is *a lot of time*.

**Comment by lildude** [comment]
> This might sound rather curt but I feel that two years is a lot of time. 😄 I'd love to help, if I can in any way.

You're right it has been a long time, but sadly it's likely to a while longer as Linguist is currently not a priority for any team within GitHub…  it is essentially running as a community maintained project with final approval from GitHub (required because it integrates directly into GitHub.com).

A change like degrouping needs to thought through thoroughly as it has been in place for a very long time and could have consequences elsewhere.

**Comment by Alhadis** [comment]
We shouldn't have to reconsider *all* grouped languages. Only the very obvious cases. Sass is no closer to CSS than CoffeeScript is to JavaScript.

**Comment by pchaigno** [comment]
@Alhadis You can if you want to start. I'll close this PR in favor of that issue in that case.

**Comment by Alhadis** [comment]
I've opened #4291 so we can discuss this matter in depth.

Let the bikeshedding begin!
